yfinance/data.py

```diff
@@ -223,6 +223,10 @@ def _get_crumb_basic(self, timeout=30):
         else:
             crumb_response = self._session.get(**get_args)
         self._crumb = crumb_response.text
+        if crumb_response.status_code == 429 or "Too Many Requests" in self._crumb:
```

Comment:
Does this change on its own solve the problem? Could revert the other changes.

Reply:
yes, this is the core change, and need to be in both places (basic and csrf). However, if bad crumb got cached somehow, it would make sense to prevent it from being used.

As for "fix spaces" - it seems there are trailing spaces in the source file so I need to change editor settings specifically to preserve those. Not sure how valuable are they for the preservation.

Reply:
How would it get cached somehow? It's deliberately never stored persistently.
Obviously keep the similar guard in `_get_crumb_csrf`, then no way to store a "429 crumb".

Reply:
if it is cached before yfinance update (e.g. requests-cache used or yfinance cache), and then used after update. But looks like it may not be needed anymore due to other changes and removal of requests-cache support.

Reply:
The `if self._session_is_caching` just before should have prevented that. Were you seeing it being reused?

Reply:
Yes, with .58 and yfinance cache(`set_tz_cache_location`) and requests-cache active I got into state when there was no attempt to update a crumb because crumb was always "Too Many Requests". I never had it with .59 version and now I'm not even sure if it was because I used Session from requests vs. curl_cffi. I removed this check as I don't think it will happen again due to proper error handling at getting crumb in the first place.